Tests/SwinjectTests/ContainerSpec.TypeForwarding.swift

```diff
@@ -129,27 +129,27 @@ class ContainerSpec_TypeForwarding: QuickSpec {
             it("resolves optional when wrapped type is registered") {
                 container.register(Dog.self) { _ in Dog() }
                 let optionalDog = container.resolve(Dog?.self)
-                let unwrappedDog = container.resolve(Dog!.self)
+                let unwrappedDog = container.resolve(Dog?.self)
```

Comment:
Ideally we do not want to lose this functionality for implicitly unwrapped optionals. Do these tests fail without `ImplicitlyUnwrappedOptional: InstanceWrapper {}` conformance?

Reply:
@jakubvano I removed the `InstanceWrapper` functionality for `ImplicitlyUnwrappedOptionals` because the compiler was complaining about the conformance as the type `ImplicitlyUnwrappedOptionals` with the following message:

```bash
Using 'ImplicitlyUnwrappedOptional' in this location is deprecated and will be removed in a future release; consider changing this to 'Optional' instead
```

And we already have the conformance for `Optional` above that part of the code:

```swift
extension Optional: InstanceWrapper {
    static var wrappedType: Any.Type { return Wrapped.self }

    init?(inContainer container: Container, withInstanceFactory factory: (() -> Any?)?) {
        self = factory?() as? Wrapped
    }
}
```

So please tell me what do you want to do here 🤔.

Reply:
I see, after reading through the [0054-abolish-iuo.md](https://github.com/apple/swift-evolution/blob/master/proposals/0054-abolish-iuo.md) it seems that implicitly unwrapped optionals will be completely removed in future swift release.

Curently it is not yet the case however and as a result of this change implicitly unwrapped optional dependencies will not be resolved (this will probably only affect SwinjectAutoregistration, see #280).

I don't see an easy way of circumventing this though, and given that it will be fixed in future swift release makes it probably not worth the effort.

Reply:
So your change is OK, although there is no point in duplicating the tests - we can remove all the`unwrappedDog` lines.

Reply:
Sounds good let me take a look at that tests